source-kafka/src/main/java/org/apache/kylin/source/kafka/KafkaInputBase.java

```diff
@@ -89,7 +95,28 @@ public DataModelDesc getDataModel() {
 
             @Override
             public List<TblColRef> getAllColumns() {
-                return flatDesc.getFactColumns();
+                final Set<TblColRef> factTableColumnSet = Sets.newHashSet();
+                TableRef rootFactTable = getDataModel().getRootFactTable();
+                for (TblColRef col : flatDesc.getAllColumns()) {
+                    if (col.getTableRef().equals(rootFactTable)) {
+                        factTableColumnSet.add(col);
+                    }
+                }
+                // Add column which belongs to root fact table in join relation but lost
+                for (JoinTableDesc joinTableDesc : getDataModel().getJoinTables()) {
+                    JoinDesc jd = joinTableDesc.getJoin();
+                    for (TblColRef colRef : jd.getForeignKeyColumns()) {
+                        if (colRef.getTableRef().equals(rootFactTable)) {
+                            factTableColumnSet.add(colRef);
+                        }
+                    }
+                    for (TblColRef colRef : jd.getPrimaryKeyColumns()) {
```

Comment:
PK columns are on lookup tables, they won't be fact table, so this loop has no necessary to be here.

Reply:
OK, I will fix it.